snuba/datasets/sessions.py

```diff
@@ -0,0 +1,259 @@
+import uuid
+from datetime import timedelta, datetime
+from typing import Mapping, Sequence, Optional
+
+from snuba.clickhouse.columns import (
+    ColumnSet,
+    DateTime,
+    LowCardinality,
+    AggregateFunction,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+)
+from snuba.datasets.dataset import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.schemas.tables import (
+    MergeTreeSchema,
+    MaterializedViewSchema,
+    AggregatingMergeTreeSchema,
+)
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.query.extensions import QueryExtension
+from snuba.query.organization_extension import OrganizationExtension
+from snuba.query.parsing import ParsingContext
+from snuba.processor import (
+    MAX_UINT32,
+    MessageProcessor,
+    ProcessedMessage,
+    ProcessorAction,
+    _collapse_uint16,
+    _collapse_uint32,
+    _ensure_valid_date,
+)
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+from snuba.query.processors.prewhere import PrewhereProcessor
+from snuba.query.project_extension import ProjectExtension, ProjectExtensionProcessor
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+WRITE_LOCAL_TABLE_NAME = "sessions_raw_local"
+WRITE_DIST_TABLE_NAME = "sessions_raw_dist"
+READ_LOCAL_TABLE_NAME = "sessions_hourly_local"
+READ_DIST_TABLE_NAME = "sessions_hourly_dist"
+READ_LOCAL_MV_NAME = "sessions_hourly_mv_local"
+READ_DIST_MV_NAME = "sessions_hourly_mv_dist"
+
+STATUS_MAPPING = {
+    "ok": 0,
+    "exited": 1,
+    "crashed": 2,
+    "abnormal": 3,
+}
+REVERSE_STATUS_MAPPING = {v: k for (k, v) in STATUS_MAPPING.items()}
+NIL_UUID = "00000000-0000-0000-0000-000000000000"
+
+
+class SessionsProcessor(MessageProcessor):
+    def process_message(self, message, metadata=None) -> Optional[ProcessedMessage]:
+        if message["duration"] is None:
+            duration = None
+        else:
+            duration = _collapse_uint32(int(message["duration"] * 1000))
+
+        # since duration is not nullable, the max duration means no duration
+        if duration is None:
+            duration = MAX_UINT32
+
+        processed = {
+            "session_id": str(uuid.UUID(message["session_id"])),
+            "distinct_id": str(uuid.UUID(message.get("distinct_id") or NIL_UUID)),
+            "seq": message["seq"],
+            "org_id": message["org_id"],
```

Comment:
We do not support org moves any more for this feature. Will document this accordingly. I will share the plan for end of project between org moves next week ideally.

Reply:
next week? That seems great, since if we actually take a decision on this I would not have to support org moves in the errors table which is already giving me nightmares